### PR TITLE
SHIPAPPS-92 upgraded to react 16

### PR DESCRIPTION
**Pull request by adamsas**

### CODE REVIEW

File: src/Typeahead.jsx [hunk 1]
```diff
@@ -410,7 +410,7 @@ Typeahead.propTypes = {
 Typeahead.defaultProps = {
   isLoading: false,
   editable: true,
-  empty: '',
+  empty: false,
```
Comment (JaredBoone) [inline src/Typeahead.jsx:413]: I see this code in a dependent project:
`      <Typeahead
        clearOnSelect
        placeholder='Search'
        empty='No projects found...'`
Thoughts @isuttell ?

Comment (JaredBoone) [reply src/Typeahead.jsx:413]: Actually, I guess that's the original non-ship-components code. False alarm.

Comment (JaredBoone) [reply src/Typeahead.jsx:413]: Seems like the "empty" string prop was moved to "placeholder" prop at some point.

File: package.json [hunk 2]
```diff
@@ -70,9 +70,8 @@
     "postcss-nested": "^1.0.0",
     "postcss-simple-vars": "^1.1.0",
     "raf": "^3.4.0",
-    "react": "^15.6.2",
-    "react-addons-css-transition-group": "^15.6.2",
-    "react-dom": "^15.6.2",
+    "react": "^16.6.3",
```
Comment (JaredBoone) [inline package.json]: @isuttell should we change these to ">=16.0.0" ?

Comment (isuttell) [reply package.json]: Same comment as before, yeah I think >= makes sense we just need to cap it off I think with <17

Comment (JaredBoone) [reply package.json]: ">=16.0.0 <17.0.0"

**Review by JaredBoone — APPROVED** [review]

**Pull request merged**